### PR TITLE
Update mitiq build badge display name

### PR DESCRIPTION
**Pull request by Misty-W**
Description:
Fixes the badge display name.

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Review by natestemen — COMMENTED** [review]
I kinda think test is actually more accurate here.

Do you know why it works when using `build` in the URL? The [docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) that Andrea linked to in the email thread on this topic suggest using the filepath of the workflow. Seeing as we don't have a workflow/file named `build`, how does this work?

**Comment by Misty-W** [comment]
> I kinda think test is actually more accurate here.
> 
> Do you know why it works when using `build` in the URL? The [docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) that Andrea linked to in the email thread on this topic suggest using the filepath of the workflow. Seeing as we don't have a workflow/file named `build`, how does this work?

We do have a workflow named `build`, see https://github.com/unitaryfund/mitiq/pull/1635/commits/774b58f0d461678fd0d71ff92cbdfcda41536789. 

It’s not a big deal but I think we should be consistent in the naming.

**Comment by natestemen** [comment]
> We do have a workflow named build, see https://github.com/unitaryfund/mitiq/commit/774b58f0d461678fd0d71ff92cbdfcda41536789.

Good point. My guess is that GitHub is being generous and allowing both the use of the filename _and_ the workflow name? Since we don't know (and seeing as the badge just broke while using the workflow name as opposed to filename)  I think we should stick with the documented path. What if we rename the workflow back to `test`?

> we should be consistent in the naming.

Agreed, consistency is always good!

**Comment by Misty-W** [comment]
GitHub is looking at the workflow `build` on the master branch, and it finds the master branch via the path specified. I think we should keep the name as `build` to keep confidence that we haven't changed the _criteria_ for a passing master branch.

**Comment by andreamari** [comment]
The confusion was caused by my PR where I introduced a difference between the workflow file and the workflow name.

@Misty-W is suggesting that, in the current situation, we should point the badge to the workflow name (`build`).
@natestemen is suggesting that it is better to use the same name for the workflow and for the file. Otherwise we are not sure what is the right name that we must use in the badge link. 

I think you are both right!
 
If I am not wrong, we have 3 options:

1. Renaming both the workflow name back to `test`,  keeping the file name to `test.yml`  and using this badge: `https://github.com/unitaryfund/mitiq/workflows/test/badge.svg?branch=master`

2.  Keeping the workflow name `build`,  renaming the file name to `build.yml`  and using this badge: `https://github.com/unitaryfund/mitiq/workflows/build/badge.svg?branch=master`

3.  Merging this PR as it is now despite using different names for the workflow and for its file, but testing that the badge works as expected. E.g. by intentionally introducing a failing test. 

All solutions work for me!

**Comment by Misty-W** [comment]
> The confusion was caused by my PR where I introduced a difference between the workflow file and the workflow name.
> 
> @Misty-W is suggesting that, in the current situation, we should point the badge to the workflow name (`build`). @natestemen is suggesting that it is better to use the same name for the workflow and for the file. Otherwise we are not sure what is the right name that we must use in the badge link.
> 
> I think you are both right!
> 
> If I am not wrong, we have 3 options:
> 
> 1. Renaming both the workflow name back to `test`,  keeping the file name to `test.yml`  and using this badge: `https://github.com/unitaryfund/mitiq/workflows/test/badge.svg?branch=master`
> 2. Keeping the workflow name `build`,  renaming the file name to `build.yml`  and using this badge: `https://github.com/unitaryfund/mitiq/workflows/build/badge.svg?branch=master`
> 3. Merging this PR as it is now despite using different names for the workflow and for its file, but testing that the badge works as expected. E.g. by intentionally introducing a failing test.
> 
> All solutions work for me!

I’m okay with solutions 2 or 3. WDYT @natestemen ?

**Comment by natestemen** [comment]
Let's go with 2, but for the sake of following the docs to the T, let's include the `.yml` in the URL as they recommend. So something like: `https://github.com/unitaryfund/mitiq/workflows/build.yml/badge.svg?branch=master`.

**Comment by natestemen** [comment]
Good catch Andrea! That's potentially another difference from the docs that could be the cause of the failure.

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
 [![All Contributors](https://img.shields.io/badge/all_contributors-34-orange.svg?style=flat-square)](#contributors-)
 <!-- ALL-CONTRIBUTORS-BADGE:END -->
-[![build](https://github.com/unitaryfund/mitiq/workflows/test/badge.svg?branch=master)](https://github.com/unitaryfund/mitiq/actions)
+[![build](https://github.com/unitaryfund/mitiq/workflows/build/badge.svg?branch=master)](https://github.com/unitaryfund/mitiq/actions)
```
Comment (Misty-W) [inline README.md:6]:
```suggestion
[![build](https://github.com/unitaryfund/mitiq/actions/workflows/build.yml/badge.svg?branch=master)
```

**Review by natestemen — APPROVED** [review]

**Pull request merged**